jekyll/_includes/snippets/faq/self-hosted-runner-faq-snip.adoc

```diff
@@ -0,0 +1,304 @@
+
+This approach is the most flexible, but will require providing the jobs sufficient privileges to install tools or install the tools in a non-overlapping manner (eg. into the working directory).
+
+* Pre-install dependencies on the machine where the machine runner is installed
+
+This approach is the most secure; however, this means that if the job’s dependencies change, the self-hosted runner machine must be reconfigured.
+
+[#what-connectivity-is-required]
+=== What connectivity is required?
+
+In order to connect back to CircleCI to receive and execute jobs, outbound HTTPS connections to `runner.circleci.com`, `circleci-binary-releases.s3.amazonaws.com` are required.
+
+NOTE: No inbound connectivity is required by a self-hosted runner. Any other required connectivity is dependent on the content of the jobs themselves.
+
+CAUTION: Using the checkout step will require access to your VCS provider. Using the cache, workspace or artifact features will require outbound HTTPS connections to `circle-production-customer-artifacts.s3.amazonaws.com`.
+
+[#how-do-caching-workspaces-and-artifacts-work-with-circleci-self-hosted-runners]
+=== How do caching, workspaces, and artifacts work with CircleCI's self-hosted runners?
+
+Caches, workspaces, and artifacts are methods you can implement to help persist data between jobs, and speed up builds. All three features are compatible with self-hosted runners.
+
+Find out more about these concepts below:
+
+* <<caching#,Caching>>
+* <<workspaces#,Workspaces>>
+* <<artifacts#,Artifacts>>
+
+You can also find out more on the <<persist-data#,Persisting data>> page.
+
+If you would prefer to take complete control of artifact storage, CircleCI recommends you avoid the built-in steps and upload the artifacts directly to your chosen storage backend.
+
+[#what-are-the-best-practices-for-managing-state-between-jobs]
+=== What are the best practices for managing state between jobs?
+
+Machine runners are unopinionated about state between jobs. Machine runners can be configured to give each job a unique working directory and clean it up afterwards - but this is optional. And by default, nothing restricts the job from placing files outside of its working directory.
+
+In general CircleCI recommends jobs rely on as little state as possible to improve their reproducibility. An effective way to accomplish this is to put cleanup steps at the start of a job so they are guaranteed to run regardless of what happened to a previous job.
+
+CAUTION: It may be possible to reduce build times by making use of caches that persist on the host between jobs, however this is a trade-off against reproducibility - and may also lead to disks filling up over time. As a result, this trade-off could lead to higher billable usage.
+
+[#how-long-do-inactive-self-hosted-runners-persist-in-the-self-hosted-runner-inventory-page]
+==== How long do inactive self-hosted runners persist in the self-hosted runner inventory page?
```

Comment:
Just checking if it's right that this is an h4 rather than h3 like the others?

Reply:
@rosieyohannan all fixed!